java/src/jmri/InstanceManager.java

```diff
@@ -233,87 +233,89 @@
         log.trace("getOptionalDefault of type {}", type.getName());
         List<T> l = getList(type);
         if (l.isEmpty()) {
+            synchronized (type) {
```

Comment:
This is synchronizing on a local variable.  You will need to synchronize on something that has static scope.